### PR TITLE
nn.Contiguous

### PR DESCRIPTION
**Pull request by szagoruyko**
Very simple module. Can be used instead of reshape to save memory when either `gradOutput` or `input` is not contiguous, and does not require size arguments.

### CODE REVIEW

File: Contiguous.lua [hunk 1]
```diff
@@ -0,0 +1,19 @@
+local Contiguous, parent = torch.class('nn.Contiguous', 'nn.Module')
+
+function Contiguous:updateOutput(input)
+   if not input:isContiguous() then
+      self.output:resizeAs(input):copy(input)
+   else
+      self.output = input
```
Comment (soumith) [inline Contiguous.lua]: can you make this to be :set, instead of =. Also, for gradInput below.

Comment (soumith) [inline Contiguous.lua]: ping!

Comment (szagoruyko) [inline Contiguous.lua]: fixed

**Comment by fmassa** [comment]
Is there some need for unit tests for this module ?

**Pull request merged**